File: src/006/p006.rs
fn solve() -> i32 {
    let (sum, sum_of_sq): (i32, i32) =
        (0..101).fold((0, 0), |acc, x| (acc.0 + x, acc.1 + x.pow(2)));
    sum.pow(2) - sum_of_sq
}

fn main() {
    let result = solve();
    println!("{}", result);
}

File: src/006/p006.ml
open Base
open Stdio

let squared x = x * x

let sum = List.fold ~init:0 ~f:(+)

let sum_of_sqs =
  List.range 1 101
  |> List.map ~f:(squared)
  |> sum

let sq_of_sum =
  List.range 1 101
  |> sum
  |> squared

let solve =
  sq_of_sum - sum_of_sqs

let () =
  printf "%d\n" solve
